delfin/drivers/huawei/oceanstor/rest_client.py

```diff
@@ -367,3 +387,202 @@ def list_alerts(self):
                                           "GET",
                                           log_filter_flag=True)
         return result_list
+
+    def _get_performance_switch(self):
+        url = "/performance_statistic_switch"
+        result = self.call(url, method='GET', log_filter_flag=True)
+
+        msg = _('Get performance_statistic_switch failed.')
+        self._assert_rest_result(result, msg)
+        self._assert_data_in_result(result, msg)
+        return result['data']
+
+    def _set_performance_switch(self, value):
+        url = "/performance_statistic_switch"
+        data = {"CMO_PERFORMANCE_SWITCH": value}
+        result = self.call(url, data, method='PUT', log_filter_flag=True)
+
+        msg = _('Set performance_statistic_switch failed.')
+        self._assert_rest_result(result, msg)
+        self._assert_data_in_result(result, msg)
+        return result['data']
+
+    def _get_performance_strategy(self):
+        url = "/performance_statistic_strategy"
+        result = self.call(url, method='GET', log_filter_flag=True)
+
+        msg = _('Get performance_statistic_strategy failed.')
+        self._assert_rest_result(result, msg)
+        self._assert_data_in_result(result, msg)
+        return result['data']
+
+    def _set_performance_strategy(self, hist_enable=1, hist_duration=60,
+                                  auto_stop=0, duration=5, max_duration=0):
+        url = "/performance_statistic_strategy"
+        data = {
+            "CMO_STATISTIC_ARCHIVE_SWITCH": hist_enable,
+            "CMO_STATISTIC_ARCHIVE_TIME": hist_duration,
+            "CMO_STATISTIC_AUTO_STOP": auto_stop,
+            "CMO_STATISTIC_INTERVAL": duration,
+            "CMO_STATISTIC_MAX_TIME": max_duration
+        }
+        result = self.call(url, data, method='PUT', log_filter_flag=True)
+
+        msg = _('Set performance_statistic_strategy failed.')
+        self._assert_rest_result(result, msg)
+        self._assert_data_in_result(result, msg)
+        return result['data']
+
+    def _get_metrics(self, resource_type, resource_id, metrics_ids):
+        url = "/performace_statistic/cur_statistic_data"
+        params = "CMO_STATISTIC_UUID={0}:{1}&CMO_STATISTIC_DATA_ID_LIST={2}&"\
+            .format(resource_type, resource_id, metrics_ids)
+        return self.paginated_call(url, None, "GET",
+                                   params=params, log_filter_flag=True)
+
+    def enable_metrics_collection(self):
+        return self._set_performance_switch('1')
+
+    def disable_metrics_collection(self):
+        return self._set_performance_switch('0')
+
+    def configure_metrics_collection(self):
```

Comment:
Why is this function performed in each collection.. this should be a prerequisite .

Reply:
This is configured as disabled by default. It is an option for user to update if needed. Run only once.